fs/fat/fs_fat32.c

```diff
@@ -764,7 +765,11 @@ static ssize_t fat_write(FAR struct file *filep, FAR const char *buffer,
 
   /* Check if the file size would exceed the range of off_t */
 
-  if (ff->ff_size + buflen < ff->ff_size)
+  const off_t off_max = (
+    sizeof(off_t) == sizeof(long long) ? LLONG_MAX :
+    sizeof(off_t) == sizeof(int) ? INT_MAX : 0);
```

Comment:
Sorry for the unclear comment. I didn't ment that moving of OFF_MAX to sys/types.h is needed, but rather to use fixed size types max value (int32_t/int64_t) instead of standard types like (int/long long). I was thinking that keeping local define is enough. If there is a need for wider visibility for OFF_MAX then we can try to follow http://fxr.watson.org/fxr/source/sys/limits.h?v=FREEBSD-9-0 and put it to sys/limits.h instead of sys/types.h

Reply:
In would be better to use fixed width integers.  'int' is suppose to represent that natural size of a register in the machine model but must be at least 16-bits in width.  On many platforms, usually 8- and 16-bit machines, it is only 16-bits wide.  On eZ80, it is 24-bits in width.  So the size of int will change size from platform to platform bit the size of off_t should not.  Better to use int32_t as it used in sys/types.h

https://en.wikipedia.org/wiki/C_data_types

int64_t is a better representation than long long too, although there is not so much variability in width.

Reply:
off_t is signed.  You do not need an OFF_MIN?